src/Compilers/CSharp/Portable/Symbols/Source/SourceModuleSymbol.cs

```diff
@@ -529,6 +533,86 @@ protected override void DecodeWellKnownAttributeImpl(ref DecodeWellKnownAttribut
             }
         }
 
+#nullable enable
+
+        internal bool RequiresRefSafetyRulesAttribute()
+        {
+            if (_lazyRequiresRefSafetyRulesAttribute == ThreeState.Unknown)
+            {
+                bool value = UseUpdatedEscapeRules && namespaceIncludesRefs(GlobalNamespace);
```

Comment:
Do we need to do anything else to handle cases where types that include refs are forwarded to this assembly?

Reply:
I'm not sure if there are additional cases we should handle here. To simplify this, we should consider emitting the attribute whenever compiling with `-langversion:11` or higher, or .NET 7 or higher. Logged [https://github.com/dotnet/roslyn/issues/63692](https://github.com/dotnet/roslyn/issues/63692).